### ISSUE TITLE
Add a Set type matching the std HashSet API

### ISSUE THREAD

**Issue by cuviper** [jonhoo/flurry#17]
Just like in `std`, it can start with the simple definition wrapping a map:

```rust
pub struct FlurryHashSet<T, S = RandomState> {
    map: FlurryHashMap<T, (), S>,
}
```

**Comment by jonhoo** [comment]
The Java code's [`KeySetView`](https://github.com/jonhoo/flurry/blob/8c014c349df1cbe03f98d84f7d59115899e4df7f/jsr166/src/ConcurrentHashMap.java#L4563) may also be of interest.

**Comment by jonhoo** [comment]
We now have a basic `HashSet` type, but it'd be great to get something along the lines of `HashMapRef` for `HashSet` too!

**Comment by GarkGarcia** [comment]
> We now have a basic `HashSet` type, but it'd be great to get something along the lines of `HashMapRef` for `HashSet` too!

Is that a case of simply wrapping `HashMapRef<T, ()>` in a newtype? I could work on that.

**Comment by twe4ked** [comment]
I _think_ you'd want to wrap `HashSet` around with `HashSetRef` similar to how `HashMap` is wrapped by `HashMapRef`.

**Comment by GarkGarcia** [comment]
> I _think_ you'd want to wrap `HashSet` with `HashSetRef` similar to how `HashMap` is wrapped by `HashMapRef`.

That's interesting. I'm not quite sure which way would be the most efficient memory wise. @jonhoo @cuviper opinions?

**Comment by GarkGarcia** [comment]
We'd be wrapping a wrapper either way...
I think wrapping `HashMap` directly should make the type's layout flatter, but I'm not sure.

**Comment by jonhoo** [comment]
Ah, no, I was imagining we just do the same trick with `HashSetRef` as with `HashMapRef`, not that we wrap `HashMapRef` to make `HashSetRef`!

**Comment by GarkGarcia** [comment]
> Ah, no, I was imagining we just do the same trick with `HashSetRef` as with `HashMapRef`, not that we wrap `HashMapRef` to make `HashSetRef`!

Makes sense.

**Comment by GarkGarcia** [comment]
@jonhoo What about the _wrapping `HashMap<T, ()>` directly_ proposal?

**Comment by jonhoo** [comment]
I think we should wrap `HashSet`, because `HashSet` actually has different _methods_ than `HashMap`, and having the wrapper _also_ do the translation from set-like methods to map-like methods (which `HashSet` is already doing) seems like unfortunate duplication.

**Comment by jonhoo** [comment]
Closed by #78!
